source/extensions/filters/http/lua/wrappers.cc

```diff
@@ -118,8 +124,38 @@ void HeaderMapWrapper::checkModifiable(lua_State* state) {
   }
 }
 
+int HeaderMapWrapper::luaSetHttp1ReasonPhrase(lua_State* state) {
+  checkModifiable(state);
+
+  const char* phrase = luaL_checkstring(state, 2);
+
+  ::Envoy::Http::StatefulHeaderKeyFormatterOptRef formatter(headers_.formatter());
+
+  if (!formatter.has_value()) {
+    using envoy::extensions::http::header_formatters::preserve_case::v3::
+        PreserveCaseFormatterConfig;
+    using ::Envoy::Http::ResponseHeaderMapImpl;
+    using ::Envoy::Http::StatefulHeaderKeyFormatter;
+    using Http::HeaderFormatters::PreserveCase::PreserveCaseHeaderFormatter;
+
+    try {
+      ResponseHeaderMapImpl& map = dynamic_cast<ResponseHeaderMapImpl&>(headers_);
+      std::unique_ptr<StatefulHeaderKeyFormatter> fmt =
+          std::make_unique<PreserveCaseHeaderFormatter>(true, PreserveCaseFormatterConfig::DEFAULT);
+      fmt->setReasonPhrase(absl::string_view(std::string(phrase)));
+      map.setFormatter(std::move(fmt));
+    } catch (const std::bad_cast&) {
+    }
```

Comment:
May be just cast this to `ResponseHeaderMapImpl*` to avoid the `try catch`.

Reply:
And I am not sure, can we just add a method to the `ResponseHeaderMap` to set reason phrase? cc @mattklein123

Reply:
It's `Http::HeaderMap` type that's being used in the `HeaderMapWrapper` class here, so unless I add the setter on it I'd still need to cast, I think (c++ is not a tool I use often, so I may be wrong).

Reply:
Switched to direct pointer cast for now.